xdsl/transforms/riscv_register_allocation.py

```diff
@@ -24,20 +78,11 @@ def allocate_registers(self, module: ModuleOp) -> None:
 
 class RegisterAllocatorBlockNaive(AbstractRegisterAllocator):
     idx: int
+    register_set: RegisterSet
 
-    def __init__(self) -> None:
+    def __init__(self, register_set: RegisterSet = _DEFAULT_REGISTER_SET) -> None:
```

Comment:
Register set is mutable, right? I'd recommend allowing None here, and creating a new default register set if None